### DISCUSSION TITLE
Auto detect the HSV green screen value

### DISCUSSION THREAD

**Discussion by timcknowles** [photobooth-app/photobooth-app#374] [Ideas]
Over the course of an event I noticed that the chroma key became less effective (green pixels showing through) at removing the background. I wondered if this was due to slightly different light levels as it got darker outside. Would it be possible to auto calibrate the HSV value for a green screen either from a test shot ( or continuously?

**Comment by mgineer85** [comment]
You have some example images from start to end?  Could remove persons for privacy.
Currently it’s only color+tolerance configurable . Maybe it needs only little adjustment … Automatic adjustment is not possible currently.

**Reply by mgineer85** [reply]
I see the section in the documentation is still a todo 😔
You could try to get the color value in an imaging program to start from the right green color and iterate from there.

**Comment by timcknowles** [comment]
Yes that’s what I did. I opened  the pi camera photo of the green screen in
Gimp and used the colour picker to select the value.

The issue was that I had done this before I arrived at the venue and I
think it was just a little off in different lightening conditions. I played
with the value and threshold on the fly and got a good result.

I was just wondering if there was a way to automate this in the future.

**Reply by mgineer85** [reply]
I do not plan to implement any automatic currently.

But feel free to send a feature request or even better a pull request if you are able to tackle this 🙂
